testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/forms/BrowserFlowTest.java

```diff
@@ -265,60 +505,74 @@ public void testSocialProvidersPresentOnLoginUsernameOnlyPageIfConfigured() {
-            execution1.setPriority(10);
-            execution1.setParentFlow(formSubflow.getId());
-            execution1 = appRealm.addAuthenticatorExecution(execution1);
+            addAuthenticatorExecution(appRealm, AuthenticationExecutionModel.Requirement.REQUIRED, UsernameFormFactory.PROVIDER_ID, 10, formSubflow);
 
             // Add REQUIRED PasswordForm Authenticator as second
-            AuthenticationExecutionModel execution2 = new AuthenticationExecutionModel();
-            execution2.setRequirement(AuthenticationExecutionModel.Requirement.REQUIRED);
-            execution2.setAuthenticatorFlow(false);
-            execution2.setAuthenticator(PasswordFormFactory.PROVIDER_ID);
-            execution2.setPriority(20);
-            execution2.setParentFlow(formSubflow.getId());
-            execution2 = appRealm.addAuthenticatorExecution(execution2);
+            addAuthenticatorExecution(appRealm, AuthenticationExecutionModel.Requirement.REQUIRED, PasswordFormFactory.PROVIDER_ID, 20, formSubflow);
 
             // Add ALTERNATIVE OTPFormAuthenticator as third
-            AuthenticationExecutionModel execution3 = new AuthenticationExecutionModel();
-            execution3.setRequirement(AuthenticationExecutionModel.Requirement.ALTERNATIVE);
-            execution3.setAuthenticatorFlow(false);
-            execution3.setAuthenticator(OTPFormAuthenticatorFactory.PROVIDER_ID);
-            execution3.setPriority(30);
-            execution3.setParentFlow(formSubflow.getId());
-            execution3 = appRealm.addAuthenticatorExecution(execution3);
+            addAuthenticatorExecution(appRealm, AuthenticationExecutionModel.Requirement.ALTERNATIVE, OTPFormAuthenticatorFactory.PROVIDER_ID, 30, formSubflow);
 
             // Add OTPForm ALTERNATIVE execution
             appRealm.setBrowserFlow(newBrowserFlow);
-        });
+        };
+    }
+
+    private static AuthenticationFlowModel copyFlow(RealmModel appRealm, String original, String newFlowAlias) {
+        AuthenticationFlowModel existingBrowserFlow = appRealm.getFlowByAlias(DefaultAuthenticationFlows.BROWSER_FLOW);
+        return AuthenticationManagementResource.copyFlow(appRealm, existingBrowserFlow, newFlowAlias);
+    }
+
+    private static AuthenticationFlowModel prepareNewBrowserForms(RealmModel appRealm, AuthenticationFlowModel newBrowserFlow, String newFlowAlias) {
+        AuthenticationFlowModel formSubFlow = appRealm.getFlowByAlias(newFlowAlias + " forms");
+        List<AuthenticationExecutionModel> executions = appRealm.getAuthenticationExecutions(formSubFlow.getId());
+
+        // Remove all executions
+        for (AuthenticationExecutionModel authExecution : executions) {
+            appRealm.removeAuthenticatorExecution(authExecution);
+        }
+
+        return formSubFlow;
     }
 
+    private static void addAuthenticatorExecution(RealmModel realm, Requirement requirement, String providerId, int priority, AuthenticationFlowModel parentFlow) {
```

Comment:
These methods are really useful, and I think that we would benefit from moving them to a util class rather than just keeping them here. For example, they could also be used to simplify the code in `KcOidcFirstBrokerLoginNewAuthTest`